enum_methods.rb

```diff
@@ -0,0 +1,146 @@
+module Enumerable
+  def my_each
+    if block_given?
+      for i in self
+        yield(i)
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    if block_given?
+      for i in self
+        yield(i, count)
+        count += 1
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) == true }
+    else
+      result = []
+      my_each { |value| result.push(value) if yield(value) == true }
+    end
+    result
+  end
+
+  def my_all?
+    count = 0
+    my_each { |value| count += 1 if yield(value) == true }
+    count == length
+  end
+
+  def my_any?
+    count = 0
+    my_each { |value| break count += 1 if yield(value) == true }
+    count.positive?
+  end
+
+  def my_none?
+    count = 0
+    my_each { |value| break count += 1 if yield(value) == true }
+    !count.positive?
+  end
+
+  def my_count
+    count = 0
+    if block_given?
+      my_each do |value|
+        count += 1 if yield(value) == true
+      end
+      count
+    else
+      puts 'No Block Given'
+    end
+  end
+
```

Comment:
- [ ] my_count returns the number of items in enum through enumeration

- [ ] my_count returns the number of items in enum through enumeration::range

- [ ] my_count counts the number of items in enum that are equal to item if an argument is given